networkx/algorithms/tests/test_core.py

```diff
@@ -67,6 +71,17 @@ def test_core_number2(self):
         assert nodes_equal(nodes_by_core[1], [1, 3])
         assert nodes_equal(nodes_by_core[2], [2, 4, 5, 6])
 
+    def test_core_number_self_loop(self):
+        self_loop = nx.cycle_graph(3)
+        self_loop.add_edge(0, 0)
+        with pytest.raises(nx.NetworkXError):
+            nx.core_number(self_loop)
+
+    def test_find_cores_warning(self):
+        G = nx.cycle_graph(3)
+        with pytest.deprecated_call():
+            nx.find_cores(G)
+
```

Comment:
As mentioned in the linked issue, this function should just be removed from the library (see #6139) so go ahead and delete this test and the warnings import.